bndtools.m2e/src/bndtools/m2e/MavenBuildRepository.java

```diff
@@ -0,0 +1,246 @@
+
+    private String getBsnFromMavenProject(MavenProject mavenProject) throws Exception {
+        final File outputDir = new File(mavenProject.getBuild().getOutputDirectory());
+
+        if (outputDir.exists()) {
+            try (Jar jar = new Jar(outputDir)) {
+                String bsn = jar.getBsn();
+
+                return bsn;
+            }
+        }
+
+        return null;
+    }
+
+    private void init() {
+        inited = true;
+
+        final IProgressMonitor monitor = new NullProgressMonitor();
+
+        for (IMavenProjectFacade projectFacade : mavenProjectRegistry.getProjects()) {
+            final IProject project = projectFacade.getProject();
+
+            try {
+                final MavenProject mavenProject = getMavenProject(projectFacade, monitor);
+
+                final String bsn = getBsnFromMavenProject(mavenProject);
+
+                if (bsn != null) {
+                    Entry<IMavenProjectFacade,MavenProject> entry = new AbstractMap.SimpleImmutableEntry<>(projectFacade, mavenProject);
+                    bsnMap.put(bsn, entry);
+                }
+            } catch (Exception e) {
+                logger.logError("Unable to get bundle symbolic name for " + project.getName(), e);
+            }
+        }
+
+        mavenProjectRegistry.addMavenProjectChangedListener(this);
+    }
+
+    private MavenProject getMavenProject(final IMavenProjectFacade projectFacade, final IProgressMonitor monitor) throws CoreException {
+        MavenProject mavenProject = projectFacade.getMavenProject();
+
+        if (mavenProject == null) {
+            mavenProject = projectFacade.getMavenProject(monitor);
+        }
+
+        return mavenProject;
+    }
+
+    @Override
+    public boolean canWrite() {
+        return false;
+    }
+
+    @Override
+    public List<String> list(String pattern) throws Exception {
+        return null;
+    }
+
+    @Override
+    public SortedSet<Version> versions(String bsn) throws Exception {
+        if (!inited) {
+            init();
+        }
+
+        List<Version> versions = new ArrayList<>();
+
+        final Entry<IMavenProjectFacade,MavenProject> entry = bsnMap.get(bsn);
+
+        if (entry != null) {
+            final MavenProject mavenProject = entry.getValue();
+            final File outputDir = new File(mavenProject.getBuild().getOutputDirectory());
+
+            if (outputDir.exists()) {
+                try (Jar jar = new Jar(outputDir)) {
+                    Version version = new Version(jar.getVersion());
+                    versions.add(version);
+                } catch (Exception e) {
+                    logger.logError("Unable to get version from " + mavenProject.getArtifactId(), e);
+                }
+            }
+        }
+
+        if (versions.isEmpty()) {
+            return SortedList.empty();
+        }
+
+        return new SortedList<Version>(versions);
+    }
+
+    @Override
+    public String getName() {
+        return "Maven Build";
```

Comment:
`Maven Workspace Repository`. The is the analog to the Workspace repo for bnd workspace builds.